### ISSUE TITLE
on_change if test for null or missing property

### ISSUE THREAD

**Issue by jjd314** [redgeoff/spiegel#161]
Allow an if clause in on_change docs like this:
```
"if": {
  "someProperty": null
}
```
to match only if the property `someProperty` is null or not present.

**Comment by redgeoff** [comment]
Hey @jjd314. This looks good, but I think I'd rather go a step further and support `sift`. Check out the sift option at https://github.com/redgeoff/spiegel/issues/122#issuecomment-391569489. It shouldn't be that much more work and it would us up to be able to perform all types of complex queries.

**Comment by jjd314** [comment]
Yes, that would be better.  I'm not going to be able to get to it soon, though.  I needed the null/missing test so I added it.  You're welcome to merge this in for a small added feature, or you can wait for sift and I'll just keep using my branch.

**Comment by redgeoff** [comment]
Cool, I think I'll wait for the sift feature before merging as it will keep the code cleaner. Thanks.

**Comment by jjd314** [comment]
Excellent, thanks!  Saves me the trouble of rebasing my branch after every update.
